pkg/controller/admissionchecks/multikueue/multikueuecluster.go

```diff
@@ -130,6 +134,53 @@ func (rc *remoteClient) queueWorkloadEvent(ctx context.Context, ev watch.Event)
 	}
 }
 
+// runGc - lists all the remote workloads having the same multikueue-origin and remove those who
+// no longer have a local corespondent (missing or awaiting deletion). If the remote workload is
+// is owned by a job,also delete the job.
+func (rc *remoteClient) runGC(ctx context.Context) {
+	log := ctrl.LoggerFrom(ctx)
+	lst := &kueue.WorkloadList{}
+	err := rc.client.List(ctx, lst, client.MatchingLabels{kueuealpha.MultiKueueOriginLabel: rc.origin})
+	if err != nil {
+		log.V(2).Error(err, "Listing remote workloads")
+		return
+	}
+
+	for _, remoteWl := range lst.Items {
+		localWl := &kueue.Workload{}
+		wlLog := log.WithValues("remoteWl", klog.KObj(&remoteWl))
+		err := rc.localClient.Get(ctx, client.ObjectKeyFromObject(&remoteWl), localWl)
+		if client.IgnoreNotFound(err) != nil {
+			wlLog.V(2).Error(err, "Reading local workload")
+			continue
+		}
+
+		if err == nil && localWl.DeletionTimestamp.IsZero() {
+			// The local workload exists and isn't being deleted, so the remote workload is still relevant.
+			continue
+		}
+
+		// if the remote wl has a controller(owning Job), delete the job
+		if controller := metav1.GetControllerOf(&remoteWl); controller != nil {
+			ownerKey := types.NamespacedName{Name: controller.Name, Namespace: remoteWl.Namespace}
+			adapterKey := schema.FromAPIVersionAndKind(controller.APIVersion, controller.Kind).String()
+			if adapter, found := adapters[adapterKey]; !found {
+				wlLog.V(2).Info("No adapter found", "adapterKey", adapterKey, "ownerKey", ownerKey)
+			} else {
+				wlLog.V(5).Info("MultiKueueGC deleting workload owner", "ownerKey", ownerKey, "ownnerKind", controller)
+				err := adapter.DeleteRemoteObject(ctx, rc.client, ownerKey)
+				if client.IgnoreNotFound(err) != nil {
+					wlLog.V(2).Error(err, "Deleting remote workload's owner", "ownerKey", ownerKey)
+				}
+			}
+		}
+		wlLog.V(5).Info("MultiKueueGC deleting remote workload")
+		if err := rc.client.Delete(ctx, &remoteWl); client.IgnoreNotFound(err) != nil {
```

Comment:
When we delete workloads despite we don't remove jobs due to the dedicated adapter is not found (Line 168), 
the job resource will be orphaned, right?